### PR TITLE
Docs: Add example notebooks

### PR DESCRIPTION
**Pull request by ameyer-rigetti**
👉 [Readable](https://qiskit-rigetti--13.org.readthedocs.build/en/13/)

### CODE REVIEW

**Comment by dbanty** [comment]
@ameyer-rigetti you might try https://github.com/spatialaudio/nbsphinx/issues/24#issuecomment-187172022 as a workaround for the "ipython3" lexer issue.

File: docs/examples/qaoa_qiskit.ipynb [hunk 1]
```diff
@@ -0,0 +1,713 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/html": [
+       "<pre style=\"word-wrap: normal;white-space: pre;background: #fff0;line-height: 1.1;font-family: &quot;Courier New&quot;,Courier,monospace\">                                  ░                              ░      »\n",
```
Comment (dbanty) [inline docs/examples/qaoa_qiskit.ipynb:235]: This graph looks much nicer in the Qiskit docs than in ours, is there a plugin or something we can use to render the symbols like they do?

### Ours
Screen Shot 2021-07-21 at 8 37 06 AM

### Theirs
Screen Shot 2021-07-21 at 8 39 19 AM

File: docs/examples/qaoa_qiskit.ipynb [hunk 2]
```diff
@@ -0,0 +1,713 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The next step is to find the optimal parameters $(\\boldsymbol{\\beta_{opt}}, \\boldsymbol{\\gamma_{opt}})$ such that the expectation value\n",
+    "\n",
+    "$$ \\langle \\psi(\\boldsymbol{\\beta}_{opt}, \\boldsymbol{\\gamma}_{opt}) \\rvert H_P \\lvert \n",
+    "\\psi(\\boldsymbol{\\beta}_{opt}, \\boldsymbol{\\gamma}_{opt}) \\rangle $$\n",
+    "\n",
+    "is minimized. Such an expectation can be obtained by doing measurement in the Z-basis. We use a classical optimization algorithm to find the optimal parameters. Following steps are involved as shown in the schematic \n",
```
Comment (dbanty) [inline docs/examples/qaoa_qiskit.ipynb:451]: This break (`\n`) doesn't seem to be working in our version, instead the schematic pushes the last line down which looks funny.

File: docs/examples/qaoa_qiskit.ipynb [hunk 3]
```diff
@@ -0,0 +1,713 @@
...
+    "\n",
+    "The code below implements the steps mentioned above."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
```
Comment (dbanty) [inline docs/examples/qaoa_qiskit.ipynb:471]: We also don't seem to be getting any Python syntax highlighting here... but I'm guessing that's the ipython lexer error?

**Review by dbanty — CHANGES_REQUESTED** [review]
There are a couple display artifacts that it would be nice to clean up in the qiskit notebook. Also, I believe that https://sphinx-copybutton.readthedocs.io/en/latest/ will work with the code blocks, we should try it!

File: docs/examples/qaoa_qiskit.ipynb (hunk 1 above)
Comment (ameyer-rigetti) [reply docs/examples/qaoa_qiskit.ipynb:235]: I noticed that. It's a nice to have, but not a priority at the moment

File: docs/examples/qaoa_qiskit.ipynb (hunk 2 above)
Comment (ameyer-rigetti) [reply docs/examples/qaoa_qiskit.ipynb:451]: Looking into it

File: docs/examples/qaoa_qiskit.ipynb (hunk 3 above)
Comment (ameyer-rigetti) [reply docs/examples/qaoa_qiskit.ipynb:472]: Fixed!

**Review by dbanty — APPROVED** [review]
LGTM 🚀

**Pull request merged**